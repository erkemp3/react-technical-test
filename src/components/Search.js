import React, { useState } from "react";
import "../styles/Search.css";
import getImages from "../requests/getImages";

const Search = ({ setSearchResults }) => {
  const [value, setValue] = useState();

  const handleSubmit = async (event) => {
    event.preventDefault();
    setSearchResults(await getImages(value));
  };

  return (
    <>
      <form className="search-form" onSubmit={handleSubmit}>
        <input
          type="text"
          className="search-form__input"
          onChange={(e) => setValue(e.target.value)}
        />
        <button className="search-form__button" type="submit">
          Search
        </button>
      </form>
    </>
  );
};

export default Search;
