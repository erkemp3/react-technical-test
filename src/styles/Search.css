.search-form {
  display: flex;
  flex-direction: row;
  margin: auto;
  margin-bottom: 30px;
}

.search-form__input {
  width: 50%;
  height: 40px;
  border-radius: 11px;
  font-size: 18px;
  font-family: "Space Mono", monospace;
  padding: 0 10px;
}

.search-form__button {
  width: 30%;
  border-radius: 11px;
  font-size: 18px;
  font-family: "Space Mono", monospace;
  padding: 0 10px;
  margin-left: 5px;
}
