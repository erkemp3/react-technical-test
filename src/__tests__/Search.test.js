import React from "react";
import { render, screen } from "@testing-library/react";
import Search from "../components/Search";

describe("Search", () => {
  const validProps = {
    setSearchResults: () => {},
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <Search setSearchResults={validProps.setSearchResults} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the input", () => {
    render(<Search setSearchResults={validProps.setSearchResults} />);
    const input = screen.getByRole("textbox");
    expect(input).toHaveAttribute("type", "text");
    expect(input).toHaveAttribute("class", "search-form__input");
  });

  it("renders the button", () => {
    render(<Search setSearchResults={validProps.setSearchResults} />);
    const button = screen.getByRole("button");
    expect(button).toHaveAttribute("type", "submit");
  });
});
