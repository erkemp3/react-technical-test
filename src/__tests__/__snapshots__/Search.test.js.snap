// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<DocumentFragment>
  <form
    class="search-form"
  >
    <input
      class="search-form__input"
      type="text"
    />
    <button
      class="search-form__button"
      type="submit"
    >
      Search
    </button>
  </form>
</DocumentFragment>
`;
